### PR TITLE
Nodemanager updating nodes to missing

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?

Updating nodemanager nodes when a chef infra node goes missing. 

### :chains: Related Resources

https://github.com/chef/automate/issues/616

### :+1: Definition of Done
When a node on the client runs page get marked missing the same node should be marked missing in the nodemanager

### :athletic_shoe: How to Build and Test the Change
1. `build components/ingest-service && build components/nodemanager-service && start_all_services`
1. Add an old CCR with` send_chef_run_example`
1. Update the missing node threshold in the node lifecycle https://a2-dev.test/settings/node-lifecycle to 6 hours. This kick off the process of updating the missing nodes.
1. Go to the client runs page to ensure the node is missing. 
1. Run the below command to see the node in the nodemanager. 
```
curl -f --insecure -H "api-token: $(get_admin_token)" https://localhost/api/v0/nodes/search -d '{
    "filters": [
      {"key": "manager_id", "values": [""]},
      {"key": "name", "values": ["insights.B"]}
    ]
  }' | jq
```
1. Ensure the "state" is "MISSING"

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: api/interservice/nodemanager/manager/manager.proto [hunk 1]
```diff
@@ -47,6 +47,7 @@ message NodeState {
 		RUNNING = 0;
 		STOPPED = 1;
 		TERMINATED = 2;
+		MISSING = 3;
```
Comment (lancewf) [inline api/interservice/nodemanager/manager/manager.proto]: Adding the new state of missing to the nodemanager

File: components/ingest-service/backend/elastic/jobs.go [hunk 2]
```diff
@@ -19,9 +19,63 @@ import (
 	"github.com/chef/automate/components/ingest-service/backend/elastic/mappings"
 )
 
+type nodeUUID struct {
+	EntityUUID string `json:"entity_uuid"`
+}
+
+// GetNodesToMarkMissing gets all the node that need updated to missing
+func (es *Backend) GetNodesToMarkMissing(ctx context.Context, threshold string) ([]string, error) {
```
Comment (lancewf) [inline components/ingest-service/backend/elastic/jobs.go]: New elasticsearch function to collect all the nodes that need marked missing.

Comment (vjeffrey) [reply components/ingest-service/backend/elastic/jobs.go]: i was gonna ask why a new function (why not just modify MarkNodesMissing to return the list of ids, but after digging in a bit I guess the BulkResponseItem/Request msg don't provide a way of doing that -- is that right?

Comment (lancewf) [reply components/ingest-service/backend/elastic/jobs.go]: You are totally right, thank you for pointing it out. I wrote this with my original thought of how this function worked but did not change my tack when I saw that it was looping through the nodes. I have updated it locally and will push it up soon.

File: components/ingest-service/backend/elastic/jobs.go [hunk 3]
```diff
@@ -127,6 +85,7 @@ func (es *Backend) MarkNodesMissing(ctx context.Context, threshold string) (int,
 					Timestamp: time.Now().UTC(),
 				})
 			bulkRequest = bulkRequest.Add(updateReq)
+			nodeIds = append(nodeIds, nodeID.EntityUUID)
```
Comment (vjeffrey) [inline components/ingest-service/backend/elastic/jobs.go]: nice! 🎉

**Pull request merged**